xarray/backends/api.py

```diff
@@ -1632,7 +1658,32 @@ def open_mfdataset(
         open_ = open_dataset
         getattr_ = getattr
 
-    datasets = [open_(p, **open_kwargs) for p in paths1d]
+    if errors not in ("raise", "warn", "ignore"):
+        raise ValueError(
+            f"'errors' must be 'raise', 'warn' or 'ignore', got '{errors}'"
+        )
+
+    datasets = []
+    remove_paths = False
+    for p in paths1d:
+        try:
+            ds = open_(p, **open_kwargs)
+            datasets.append(ds)
+        except Exception as e:
+            # remove invalid paths
+            if combine == "nested":
+                paths = _remove_path(paths, p)
+                remove_paths = True
+            if errors == "raise":
+                raise
+            if errors == "warn":
+                warnings.warn(
+                    f"Could not open {p} due to {e}. Ignoring.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+            continue
```

Comment:
I think this would be a bit less involved. We can use `emit_user_level_warning` here, too.


```suggestion
        except Exception as e:
            if errors == "raise":
                raise
            elif errors == "warn":
                emit_user_level_warnings(f"Could not open {p} due to {e}. Ignoring.")
            # remove invalid paths
            if combine == "nested":
                paths = _remove_path(paths, p)
                remove_paths = True
```